systemd_info - extend support to timer unit

NomakCooper opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
This PR extends **`systemd_info`** support to **`timer`** units. 

Example:
```yaml
- name: Gather info of systemd-tmpfiles-clean.timer and extra AccuracyUSec
  community.general.systemd_info:
    unitname:
      - systemd-tmpfiles-clean.timer
    extra_properties:
      - AccuracyUSec
  register: results
```

In addition, it also fulfills the requirements outlined in the description of Feature Idea #4896.

- added timers to **`determine_category()`**
- added timers to **`get_category_base_props()`**
- added timers to **`unit_types[]`**
- updated the main **`description`** and the one for **`unitname`**
- added new steps to the **`integration`** test tasks

Additionally, I wanted to update the description to clarify how the property values are collected, in order to avoid any misunderstandings for users.

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
Fixes #4896
<!--- Please do not forget to include a changelog fragment:
      https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-changelog-fragments
      No need to include one for docs-only or test-only PR, and for new plugin/module PRs.
      Read about more details in CONTRIBUTING.md.
      -->

##### ISSUE TYPE
<!--- Pick one or more below and delete the rest.
      'Test Pull Request' is for PRs that add/extend tests without code changes. -->
- Feature Pull Request

##### COMPONENT NAME
<!--- Write the SHORT NAME of the module, plugin, task or feature below. -->
plugins/modules/systemd_info.py
##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

russoz commented on plugins/modules/systemd_info.py:
```diff
@@ -13,9 +13,10 @@
 short_description: Gather C(systemd) unit info
 description:
-  - This module gathers info about systemd units (services, targets, sockets, mount).
+  - This module gathers info about systemd units (services, targets, sockets, mount, timer).
```
Grammar nit pick: the plurals and singulars should not mix, IMHO. Use either:
```suggestion
  - This module gathers info about systemd units (services, targets, sockets, mounts, timers).
```
or
```suggestion
  - This module gathers info about systemd units types (service, target, socket, mount, timer).
```
or something like that. Hopefully you get the idea.

felixfontein replied:
I'd also mention that support for timer was added in community.general 10.5.0. (Probably somewhere else in the description.)

russoz commented on plugins/modules/systemd_info.py:
```diff
@@ -275,7 +288,8 @@ def get_category_base_props(category):
         'socket': ['FragmentPath', 'UnitFileState', 'UnitFilePreset'],
-        'mount': ['Where', 'What', 'Options', 'Type']
+        'mount': ['Where', 'What', 'Options', 'Type'],
+        'timer': ['FragmentPath', 'UnitFileState', 'UnitFilePreset']
```
We missed that in the `mount` line, so now it has been changed (unnecessarily, we might argue). So, let's do it for the next time ;-)
```suggestion
        'timer': ['FragmentPath', 'UnitFileState', 'UnitFilePreset'],
```

russoz reviewed (commented):
Hi @NomakCooper thanks for contributing again! :-)

Two minor nits, otherwise it LGTM

NomakCooper commented:
@russoz @felixfontein thanks as always for the comments. :smiley:

- Fixed the unit types in the main description.
- Fixed the timer line in **`base_props{}`**
- Added an additional note in the main description specifying that **`timers`** are supported from **`community.general 10.5.0`**

russoz reviewed (approved):
LGTM

felixfontein commented:
@NomakCooper thanks for your contribution!
@russoz thanks for reviewing!
